### PR TITLE
chore: remove .html extensions from md links

### PR DESCRIPTION
**Pull request by wescopeland**
Follow-up to #120.

### CODE REVIEW

File: docs/orphaned/free-games.md
Comment (fomdt) [inline docs/orphaned/free-games.md:1]: All changed links in this file are external.

File: docs/developer-docs/achievement-design.md [hunk 1]
```diff
@@ -279,6 +279,6 @@ Challenge Achievements are your chance to show players how to play their favorit
 
 ## External Resources
 
-- [Best practices achievement creation guide](http://www.voidcn.com/article/p-gmokauqn-bgh.html) written by Jeff Sullivan, Developer Account Manager of XNA Developer Connection (XDC) regarding XBox 360 Achievements.
+- [Best practices achievement creation guide](http://www.voidcn.com/article/p-gmokauqn-bgh) written by Jeff Sullivan, Developer Account Manager of XNA Developer Connection (XDC) regarding XBox 360 Achievements.
```
Comment (fomdt) [inline docs/developer-docs/achievement-design.md]: External link, but it appears dead any way.

File: docs/es/developer-docs/achievement-design.md [hunk 2]
```diff
@@ -274,6 +274,6 @@ Logros sobre retos son tu oportunidad para mostrarle a los jugadores como jugar
 
 ## Recursos Externos
 
-- [Guía en las Mejores practicas en la creación de logros](http://www.voidcn.com/article/p-gmokauqn-bgh.html) escrito por Jeff Sullivan, Desarrollador en Administración de Cuentas de XNA Developer Connection (XDC) respecto a los logros de Xbox 360.
+- [Guía en las Mejores practicas en la creación de logros](http://www.voidcn.com/article/p-gmokauqn-bgh) escrito por Jeff Sullivan, Desarrollador en Administración de Cuentas de XNA Developer Connection (XDC) respecto a los logros de Xbox 360.
```
Comment (fomdt) [inline docs/es/developer-docs/achievement-design.md]: External link, but it appears dead any way.

File: docs/pt/developer-docs/console-specific-tips.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ Determinadas variedades de valores significam coisas diferentes no contexto do h
 - ❌ **0xFE00 - 0xFE9F: Mais dados gráficos** (igual aos dados gráficos acima)
 - ❌ **0xFEA0 - 0xFFFF: Diversos** (não use)
 
-Para mais informações, consulte: [http://gameboy.mongenel.com/dmg/asmmemmap.html](http://gameboy.mongenel.com/dmg/asmmemmap.html) `texto em inglês`
+Para mais informações, consulte: [http://gameboy.mongenel.com/dmg/asmmemmap.html](http://gameboy.mongenel.com/dmg/asmmemmap) `texto em inglês`
```
Comment (fomdt) [inline docs/pt/developer-docs/console-specific-tips.md]:
```suggestion
Para mais informações, consulte: [http://gameboy.mongenel.com/dmg/asmmemmap.html](http://gameboy.mongenel.com/dmg/asmmemmap.html) `texto em inglês`
```

File: docs/pt/developer-docs/achievement-design.md [hunk 4]
```diff
@@ -275,6 +275,6 @@ As conquistas de desafio são a sua chance de mostrar aos jogadores como jogar s
 
 ## Fontes externas
 
-- [Best practices achievement creation guide](http://www.voidcn.com/article/p-gmokauqn-bgh.html) escrito por Jeff Sullivan, gerente de contas de desenvolvedor do XNA Developer Connection (XDC) sobre Conquistas do XBox 360. `texto em inglês`
+- [Best practices achievement creation guide](http://www.voidcn.com/article/p-gmokauqn-bgh) escrito por Jeff Sullivan, gerente de contas de desenvolvedor do XNA Developer Connection (XDC) sobre Conquistas do XBox 360. `texto em inglês`
```
Comment (fomdt) [inline docs/pt/developer-docs/achievement-design.md]: External link, but it appears dead any way.

File: docs/es/general/contributing-to-the-docs.md [hunk 5]
```diff
@@ -186,15 +186,15 @@ Los enlaces internos deben seguir esta sintaxis:
...
+[Consejos y trucos](/developer-docs/tips-and-tricks)
 ```
 
 No uses rutas absolutas al referenciar otra página en la documentación. VitePress detectará automáticamente y advertirá cuando los enlaces estén rotos, pero esta detección no se activa para rutas absolutas.
 
 ```md
 🛑 MAL
 
-[Consejos y trucos](https://docs.retroachievements.org/developer-docs/tips-and-tricks.html)
+[Consejos y trucos](https://docs.retroachievements.org/developer-docs/tips-and-tricks)
```
Comment (fomdt) [inline docs/es/general/contributing-to-the-docs.md]:
```suggestion
[Consejos y trucos](https://docs.retroachievements.org/developer-docs/tips-and-tricks.html)
```

File: docs/general/contributing-to-the-docs.md [hunk 6]
```diff
@@ -164,15 +164,15 @@ Internal docs links should follow this syntax:
 ```
 ✅ GOOD
 
-[Tips and Tricks](/developer-docs/tips-and-tricks.html)
+[Tips and Tricks](/developer-docs/tips-and-tricks)
```
Comment (fomdt) [inline docs/general/contributing-to-the-docs.md:167]:
```suggestion
[Tips and Tricks](/developer-docs/tips-and-tricks)

[Rules](/guidelines/users/code-of-conduct#rules)
```

File: docs/general/contributing-to-the-docs.md [hunk 7]
```diff
@@ -164,15 +164,15 @@ Internal docs links should follow this syntax:
...
 ```
 
 Do **NOT** use absolute paths when referencing another page on the docs. VitePress will automatically detect and warn when links are broken, but this detection doesn't kick in for absolute paths.
 
 ```
 🛑 BAD
 
-[Tips and Tricks](https://docs.retroachievements.org/developer-docs/tips-and-tricks.html)
+[Tips and Tricks](https://docs.retroachievements.org/developer-docs/tips-and-tricks)
```
Comment (fomdt) [inline docs/general/contributing-to-the-docs.md]:
```suggestion
[Tips and Tricks](https://docs.retroachievements.org/developer-docs/tips-and-tricks.html)

[Rules](https://docs.retroachievements.org/guidelines/users/code-of-conduct.html#rules)
```

File: docs/es/general/contributing-to-the-docs.md (hunk 5 above)
Comment (wescopeland) [reply docs/es/general/contributing-to-the-docs.md]: Better to use a relative link here so dead link detection works properly.

Comment (wescopeland) [reply docs/es/general/contributing-to-the-docs.md]: Ah, I misinterpreted. This is covering a bad practice.

Comment (fomdt) [reply docs/es/general/contributing-to-the-docs.md]: To be more specific, this example follows a sentence about not using absolute paths when referencing another page on the docs, [English documentation for reference](https://docs.retroachievements.org/general/contributing-to-the-docs.html#links).

Relative links with extensions are not explicitly shown as bad, but if they should be, the best place in this document would be between lines 190 and 192 in my opinion.

**Pull request merged**